src/main/scala/org/broadinstitute/hail/driver/FilterAlleles.scala

```diff
@@ -126,29 +144,58 @@ object FilterAlleles extends Command {
 
       def downcodeGenotype(g: Genotype): Genotype = {
         val px = g.px.map(downcodePx)
-        val gq = px.map(Genotype.gqFromPL)
-        Genotype(g.gt.map(downcodeGt),
-          g.ad.map(downcodeAd),
-          g.dp,
-          gq,
-          px,
-          g.fakeRef,
-          g.isDosage
+        g.copy(gt = g.gt.map(downcodeGt),
+          ad = g.ad.map(downcodeAd),
+          gq = px.map(Genotype.gqFromPL),
+          px = px
+        )
+      }
+
+      def truncatePx(px: Array[Int]): Array[Int] = {
```

Comment:
I think code maintenance is easier if we keep the user facing name `--subset` consistent with the code, i.e. rename this to `subsetPx`.